ui/core/src/model/calculations.ts

```diff
@@ -55,56 +56,151 @@ export const CALCULATIONS_CONFIG: Readonly<Record<CalculationType, CalculationCo
     label: 'Sum',
     description: 'The sum of all values',
   },
+  Min: {
+    label: 'Min',
+    description: 'Minimum value',
+  },
+  Max: {
+    label: 'Max',
+    description: 'Maximum value',
+  },
 } as const;
 
 export const DEFAULT_CALCULATION: CalculationType = 'Sum';
 
 type CalculationValue = number | null | undefined;
 
+/**
+ * Calculate a multiple values for a set of time series data.
+ *
+ * @param values - Array of time series data.
+ * @param includeCalculations - Array of calculations to include.
+ */
+export function getCalculations<IncludeCalcs extends CalculationType[]>(
```

Comment:
I'm hoping this is relatively readable with how I've written and commented it, but please let me know if it's confusing. Cases where we iterate over a ton of data are probably good ones to err a little more on perf optimizations over readability, but I want to make sure it's maintainable.

Reply:
💯 I was debating making it a little more generic, but decided not to in this PR because it was already big enough, but could totally see us doing so the next time we need to touch this.